### PR TITLE
feat: Block AI scraping bots

### LINKED ISSUE #3850: Block AI scraping bots
### Motivation

In the past few months, we have seen AI scraping bots become more and more prevalent, especially ClaudeBot. Personally, I've seen it do as much as 10K requests in 24 hours, with some reporting as many as [2 million requests in 7 days](https://www.reddit.com/r/Anthropic/comments/1c8tu5u/comment/l17430p/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button). Since regular anti scraping/crawling bots rules are implemented in CRS, it isn't such a far fetched idea to implement one specifically for AI scrapers.

This would:
1. Reduce the amount of traffic being actually processed
2. Allow end-users to prevent AI bots from stealing their data
3. Prevent attackers from using AI scraping bots to automate their enumeration and pentesting efforts

### Proposed solution

Add a `SecRule` to `REQUEST-913-SCANNER-DETECTION.conf`, along side a `.data` file containing the User-agents of the AI scraping bots.

The rule should be separate rule to differentiate from the other non-AI crawler rule trigger messages. 

### Alternatives

An alternative solution would be adding to the `robots.txt` file to disallow these bots. Unfortunately, it has been seen that they ignore the `robots.txt` file.

### PR DESCRIPTION
**Pull request by highpingblorg**
_(Fixes #3850)_

Pull request for issue #3850.

The pull request is still missing tests for the added rule. I've looked through the documentation for testing and I can't seem to find the appropriate way to test for a rule that adds new data, in this case a list of AI scraper User-agents.

Any help and advice is always welcome.

### CODE REVIEW

**Comment by fzipi** [comment]
Thanks for your contribution @highpingblorg!

Writing tests is easy. Maybe taking a quick look at our https://github.com/coreruleset/coreruleset/blob/main/CONTRIBUTING.md, will give you more insights. Here is the Tl; DR version https://github.com/coreruleset/coreruleset/blob/main/CONTRIBUTING.md#writing-tests.

See for example test id 6 [here](https://github.com/coreruleset/coreruleset/blob/316e196cf8beebe66056c1f92bd37547d9f7e76d/tests/regression/tests/REQUEST-913-SCANNER-DETECTION/913100.yaml#L98-L113). You can copy the 913100.yaml file as 913110.yaml and then modify one or two tests to match your input on user-agents.

File: rules/REQUEST-913-SCANNER-DETECTION.conf [hunk 1]
```diff
@@ -55,6 +55,26 @@ SecRule REQUEST_HEADERS:User-Agent "@pmFromFile scanners-user-agents.data" \
     severity:'CRITICAL',\
     setvar:'tx.inbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
 
+SecRule REQUEST_HEADERS:User-Agent "@pmFromFile ai-scrapers-user-agents.data" \
+    "id:913101,\
```
Comment (fzipi) [inline rules/REQUEST-913-SCANNER-DETECTION.conf:59]:
```suggestion
    "id:913110,\
```

Per https://github.com/coreruleset/coreruleset/blob/main/CONTRIBUTING.md#id-numbering-scheme, let's use `913110` as id.

File: rules/REQUEST-913-SCANNER-DETECTION.conf [hunk 2]
```diff
@@ -55,6 +55,26 @@ SecRule REQUEST_HEADERS:User-Agent "@pmFromFile scanners-user-agents.data" \
...
+    phase:1,\
+    block,\
+    capture,\
+    t:none,\
+    msg:'Found User-Agent associated with AI scraper',\
+    logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}: %{MATCHED_VAR}',\
+    tag:'application-multi',\
+    tag:'language-multi',\
+    tag:'platform-multi',\
+    tag:'attack-reputation-scanner',\
```
Comment (fzipi) [inline rules/REQUEST-913-SCANNER-DETECTION.conf:69]:
```suggestion
    tag:'attack-ai-scraping-bot',\
```

File: rules/ai-scrapers-user-agets.data [hunk 3]
```diff
@@ -0,0 +1,28 @@
+
+# AI crawler/scraper bots
+# https://www.foundationwebdev.com/2023/11/which-web-crawlers-are-associated-with-ai-crawlers/
+# https://www.anthropic.com/
```
Comment (fzipi) [inline rules/ai-scrapers-user-agets.data:4]:
```suggestion
# AI crawler/scraper bots
# Sources:
# - https://www.foundationwebdev.com/2023/11/which-web-crawlers-are-associated-with-ai-crawlers/
# - https://www.anthropic.com/

```

**Review by fzipi — COMMENTED** [review]
Looks good! Some small changes.

**Comment by dune73** [comment]
Thank you for your contribution @highpingblorg.

I see the problem, yet I am skeptical about the proposed solution (like @RedXanadu in the original issue).

For me, this is another iteration of misbehaving bots, nothing really specific AI about it. And it's not really the fact they are exploring the site, but the rate of requests. So what you really need is rate limiting and CRS is just a weak tool that attempts to block them all when the truly malicious ones will simply change their UA.

That's why we removed all these UA based rules beyond the most striking security scanners.

When trying to update the crawler, scanner, bot and what not rules to v4, I assembled a list of around 2K user-agents of dubious renown. The difficulty is getting an overview and doing some useful taxonomy. Lacking that we opted to remove.

But this does not rule out a plugin attempting just that. All it takes somebody to do that and this PR could be a first step. Personally, I would not try to handle AI bots differently from all the other ones. But I might be wrong on this question.

As for this PR, please don't merge before we discussed this in the team.

**Review by dune73 — CHANGES_REQUESTED** [review]
See https://github.com/coreruleset/coreruleset/pull/3852#issuecomment-2404184049

**Pull request closed without merging**